### ISSUE TITLE
Support 256bit keys

### ISSUE THREAD

**Issue by ariselseng** [berlincode/sjcl#3]
Is there any way we can have these options work out of the box with sjcl? 
https://github.com/nextcloud/passman/blob/master/docs/api.md#decrypting-credentials--challenge-password

**Comment by nodauf** [comment]
I had the same issue:
Apparently the repo is the maintain. If someone else has the save error. I have fix it but modified the file https://github.com/berlincode/sjcl/blob/master/sjcl/sjcl.py line 113
I had replace
`if dkLen != 16:`
by
`if dkLen != 16 and dkLen != 32:`

There is also some issue with the padding so you need to use the following function on the field:
```
         # Correct padding
         if len(data["salt"]) % 4:
         # not a multiple of 4, add padding:
             data["salt"] += '=' * (4 - len(data["salt"]) % 4)
```

I hope this thing can help someone else.
The error I had was:
`Exception: key length should be 16 bytes`
and
`binascii.Error: Incorrect padding`

**Comment by ps-george** [comment]
This was solved by #5 

Issue can be closed.

**Comment by berlincode** [comment]
Closing this.
